pkg/network/usm/monitor.go

```diff
@@ -194,3 +242,81 @@ func (m *Monitor) Stop() {
 func (m *Monitor) DumpMaps(maps ...string) (string, error) {
 	return m.ebpfProgram.DumpMaps(maps...)
 }
+
+func (m *Monitor) getHTTP2EBPFTelemetry() *netebpf.HTTP2Telemetry {
+	var zero uint64
+	mp, _, err := m.ebpfProgram.Manager.Manager.GetMap(probes.HTTP2TelemetryMap)
+	if err != nil {
+		log.Warnf("error retrieving http2 telemetry map: %s", err)
+		return nil
+	}
+
+	http2Telemetry := &netebpf.HTTP2Telemetry{}
+	if err := mp.Lookup(unsafe.Pointer(&zero), unsafe.Pointer(http2Telemetry)); err != nil {
+		// This can happen if we haven't initialized the telemetry object yet
+		// so let's just use a trace log
+		if log.ShouldLog(seelog.TraceLvl) {
+			log.Tracef("error retrieving the http2 telemetry struct: %s", err)
+		}
+		return nil
+	}
+	return http2Telemetry
+}
+
+// Describe returns all descriptions of the collector
+func (m *Monitor) Describe(ch chan<- *prometheus.Desc) {
+	ch <- MonitorTelemetry.EndOfStreamEOS
+	ch <- MonitorTelemetry.EndOfStreamRST
+	ch <- MonitorTelemetry.StrLenGraterThenFrameLoc
+	ch <- MonitorTelemetry.StrLenTooBigMid
+	ch <- MonitorTelemetry.StrLenTooBigLarge
+	ch <- MonitorTelemetry.RequestSeen
+	ch <- MonitorTelemetry.ResponseSeen
+	ch <- MonitorTelemetry.FrameRemainder
+	ch <- MonitorTelemetry.MaxFramesInPacket
+
+}
+
+// Collect returns the current state of all metrics of the collector
```

Comment:
Generally, i'm not sure we should follow that part of NPM telemetry 
Pedro have done a great job to introduce internal telemetry, I think you should use his approach
for reference you can take a look in Daniel's implementation over Pedro's changes
https://github.com/DataDog/datadog-agent/pull/20378